Kotlin Setup

manijshrestha opened the pull request:
Wecome Kotlin!

manijshrestha committed:
Configured Kotlin

manijshrestha committed:
Added as Sourceset

manijshrestha committed:
convert to lateinit

manijshrestha committed:
converted presenter to kotlin

cdflynn commented on app/src/main/kotlin/zeta/android/apps/presenter/NavigationPresenter.kt:
```diff
@@ -0,0 +1,42 @@
+package zeta.android.apps.presenter
```
Kotlin!

cdflynn commented on app/src/main/kotlin/zeta/android/apps/ui/activity/NavigationActivity.kt:
```diff
@@ -0,0 +1,190 @@
+
+    @Inject lateinit var mPresenter: NavigationPresenter
+
+    internal class Views(root: AppCompatActivity) : BaseViews(root.findViewById(R.id.zeta_drawer_layout)) {
```
Whoa, that's neat.  Is the `: BaseViews()` defining the constructor for this class, as opposed to using `constructor //...`?  There's a lot I need to learn about Kotlin.

manijshrestha replied:
:) Yea, as time permits I will try to update this project with some kotlin goodies

cdflynn commented on app/src/main/kotlin/zeta/android/apps/ui/activity/NavigationActivity.kt:
```diff
@@ -0,0 +1,190 @@
+    }
+
+    override fun onBackPressed() {
+        if (mViews != null && mViews!!.drawerLayout.isDrawerOpen(GravityCompat.START)) {
```
does it work to do `mViews?.drawerLayout.isDrawerOpen(GravityCompat.START) ?: false`
Pardon my unfamiliarity with the allowed syntax.

manijshrestha replied:
yep good catch. I just used the built in "Convert to Kotlin class" from IDE here. it doesn't make sense to do `!!`. I will update this in later commit.

cdflynn commented on dependencies.gradle:
```diff
@@ -1,11 +1,11 @@
 ext.versions = [
         compileSdk                        : 25,
-        minSdk                            : 16,
+        minSdk                            : 19,
```
👍

cdflynn reviewed (approved):
@manijshrestha @manjunathc23 
Hope you guys are having fun at IO!  I'm so happy Kotlin is now an official Android language.

Cheers

manjunathc23 commented:
@manijshrestha  : Looks like build error :)

manjunathc23 commented:
Should we do another repo just for kotlin? @manijshrestha @cdflynn

manjunathc23 commented on build.gradle:
```diff
@@ -1,8 +1,5 @@
 import com.android.build.gradle.internal.tasks.AndroidTestTask
 
-// Top-level build file where you can add configuration options common to all sub-projects/modules.
-apply from: 'dependencies.gradle'
```
Is this not required ?

manijshrestha replied:
Yep it is no longer required. Sorry haven't had time to update this PR.

manijshrestha committed:
Merge branch 'master' into ms_kotlin

# Conflicts:
#	app/build.gradle
#	build.gradle
#	dependencies.gradle

manijshrestha committed:
Up the gradle version

manijshrestha committed:
Upgraded to Dagger 2.11
Updated the module to grab dependency using new annotation
Use kapt over annotationProcessor

manijshrestha committed:
Use alpha7

manijshrestha committed:
remove warning

manijshrestha committed:
fixed vector path warning

manijshrestha committed:
dagger progurad rules

manijshrestha committed:
Delete ci.sh

Deleted circle CI script

manijshrestha committed:
Delete circle.yml

Delete circle yml as we no longer will use circle ci

manijshrestha committed:
Revert "Delete ci.sh"

This reverts commit 339f861d653cd7313c13f8ed0d7c412d2a08cfe6.

manjunathc23 commented on ci.sh:
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+# Please run it from root project directory
+./gradlew clean build -PdisablePreDex -PwithDexcount
```
👍

manjunathc23 reviewed (commented):
LGTM

manjunathc23 commented:
@manijshrestha : I am getting lot of rebase conflicts. Can you please rebase this against master whenever you are free?

manijshrestha commented:
@manjunathc23 , I have already merged master to this branch. Try merging instead of rebasing.

manjunathc23 commented:
This was merged, Thank you @manijshrestha
